e2etests/tests/routes.go

```diff
@@ -288,4 +290,147 @@ var _ = Describe("Routes between bgp and the fabric", Ordered, func() {
+			ns := corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: testNamespace}}
+			_, err := cs.CoreV1().Namespaces().Create(context.Background(), &ns, metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(func() error {
+				_, err := cs.CoreV1().Namespaces().Get(context.Background(), testNamespace, metav1.GetOptions{})
+				return err
+			}, 2*time.Minute, time.Second).ShouldNot(HaveOccurred())
+
+			By("Creating the test pod")
+			testPod, err = k8s.CreateAgnhostPod(cs, "test-pod", testNamespace)
+			Expect(err).NotTo(HaveOccurred())
+			testPodName := testPod.Name
+
+			By("Waiting for the test pod to be ready")
+			Eventually(func() error {
+				var err error
+				testPod, err = cs.CoreV1().Pods(testNamespace).Get(context.Background(), testPodName, metav1.GetOptions{})
+				if err != nil {
+					return err
+				}
+				if !k8s.PodIsReady(testPod) {
+					return fmt.Errorf("pod %s is not ready", testPod.Name)
+				}
+				return nil
+			}, 2*time.Minute, time.Second).ShouldNot(HaveOccurred())
+
+			podNode, err = cs.CoreV1().Nodes().Get(context.Background(), testPod.Spec.NodeName, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			nodeSelector := k8s.NodeSelectorForPod(testPod)
+
+			By("Creating the frr-k8s configuration for the node where the test pod runs and advertising the pod ip")
+			frrK8sConfigRedForPod, err := frrk8s.ConfigFromVNI(vniRed, frrk8s.WithNodeSelector(nodeSelector), frrk8s.AdvertisePrefixes(testPod.Status.PodIP+"/32"))
+			Expect(err).NotTo(HaveOccurred())
+			frrK8sConfigBlueForPod, err := frrk8s.ConfigFromVNI(vniBlue, frrk8s.WithNodeSelector(nodeSelector), frrk8s.AdvertisePrefixes(testPod.Status.PodIP+"/32"))
+			Expect(err).NotTo(HaveOccurred())
+
+			err = Updater.Update(config.Resources{
+				VNIs: []v1alpha1.VNI{
+					vniRed,
+					vniBlue,
+				},
+				FRRConfigurations: []frrk8sv1beta1.FRRConfiguration{
+					frrK8sConfigRedForPod,
+					frrK8sConfigBlueForPod,
+				},
+			})
+			Expect(err).NotTo(HaveOccurred())
+
+			frrK8sPodOnNode, err := frrk8s.PodForNode(cs, testPod.Spec.NodeName)
+			validateFRRK8sSessionForVNI(vniRed, Established, frrK8sPodOnNode)
+			validateFRRK8sSessionForVNI(vniBlue, Established, frrK8sPodOnNode)
+
+			By("setting redistribute connected on leaves")
+			redistributeConnectedForLeaf(infra.LeafAConfig)
+			redistributeConnectedForLeaf(infra.LeafBConfig)
+		})
+
+		AfterEach(func() {
+			By("Deleting the test namespace")
+			err := cs.CoreV1().Namespaces().Delete(context.Background(), testNamespace, metav1.DeleteOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(func() bool {
+				_, err := cs.CoreV1().Namespaces().Get(context.Background(), testNamespace, metav1.GetOptions{})
+				return errors.IsNotFound(err)
+			}, time.Minute, time.Second).Should(BeTrue())
+		})
+
+		It("should be able to reach the hosts from the test pod and vice versa", func() {
+			tests := []struct {
+				vni      v1alpha1.VNI
+				hostName string
+				hostIP   string
```

Comment:
can we have a target/external prefix or similar here? makes it easier to follow what's hostSide vs others

Reply:
not sure I understood correctly, I am going to rename this to externalHost assuming this is your intention